### DISCUSSION TITLE
Error 401 Unauthorized

### DISCUSSION THREAD

**Discussion by LukasSchneider1** [deta/cloud-docs#446] [Help]
I recently set up a Base in Deta. Then i tried accessing the data with python what worked. Now i am trying to access the Data through the http api mentioned in the Deta Docs. I followed the instructions step by step but if i try to get the Data i get the Error that i am Unauthorized. I checked if i used the right ProjectKey and ProjectID. Everything was fine. Also the URL was right.

This is my code. 
I have an html file with buttons and a Java Script file where this functions are. 

function registerEvents() {
  document.getElementById("loadTodos").addEventListener("click", function () {
    loadTodos();
  });
}

registerEvents();

function loadTodos() {
  fetch("https://database.deta.sh/v1/MyProjectID/MyBaseName", {
    headers: {
      "X-API-Key": "My_ProjectKey",
      "Content-type": "application/json"      
    },
    
  })
    .then(function (response) {
      return response.json();
    })
    .then(function (data) {
      console.log(data);
    })
    .catch(function (err) {
      console.log(err);
    });
}

Do i need to create a Micro as a Proxy or what would you suggest me to do?
Thanks for your help.

**Comment by jnsougata** [comment]
which method you tried? getting data by key or something? cause the endpoint you are using is `root`

**Reply by LukasSchneider1** [reply]
I tested this method:
https://database.deta.sh/v1/MyProjectID/MyBaseName/Key

and this:
https://database.deta.sh/v1/MyProjectID/MyBaseName/items/Key

and nothing worked.

**Comment by abdelhai** [comment]
1. please don't use base in the frontend, it is not secure as your project key is exposed and your data can be stolen and altered
2. if you know what you are doing, use the deta-javascript sdk
